conmon/conmon.c

```diff
@@ -582,18 +694,28 @@ int main(int argc, char *argv[])
 					pipe = STDOUT_PIPE;
 				else if (masterfd == masterfd_stderr)
 					pipe = STDERR_PIPE;
+				else if (oom_handling_enabled && masterfd == efd) {
+					if (read(efd, &oom_event, sizeof(uint64_t)) != sizeof(uint64_t))
+						nwarn("read eventfd");
+					ninfo("OOM received");
+					if (open("oom", O_CREAT, 0666) < 0) {
+						nwarn("failed to write oom file");
+					}
+				}
 				else {
 					nwarn("unknown pipe fd");
 					goto out;
 				}
 
-				num_read = read(masterfd, buf, BUF_SIZE);
-				if (num_read <= 0)
-					goto out;
+				if (masterfd == masterfd_stdout || masterfd == masterfd_stderr) {
+					num_read = read(masterfd, buf, BUF_SIZE);
+					if (num_read <= 0)
+						goto out;
```

Comment:
goto is used here as we are jumping out of a nested loop (one of the few places where it is acceptable). Feel free to create a PR to fix that separately if you want :)